libmcount/trigger.c

```diff
@@ -0,0 +1,114 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <unistd.h>
+#include <string.h>
+#include <signal.h>
+#include <sys/time.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "libmcount/mcount.h"
+#include "utils/env-file.h"
+#include "utils/utils.h"
+
+
+#define ERROR_LIBRARY_NOT_EXIST \
+"%s does not exist libmcount shared object.\n"				\
+"\tmake sure that the file exists.\n\n"					\
+"\t[notification]\n"							\
+"\twhen you use dynamic tracing to the process, you must specify the\n"	\
+"\tabsolute path of 'libmcount-dynamic.so' not the relative path.\n\n" 	\
```

Comment:
Where comes this restriction from?  Isn't it just a matter of calling `realpath()`?

Reply:
if we can assume that uftrace has been installed, we can get path of libmcount-dynamic.so. otherwise, libtrigger.so has the path as default from attached process located at. so, we can get the absolute path of libmcount-dynamic.so.